mesonbuild/backend/ninjabackend.py

```diff
@@ -374,6 +374,10 @@ def write(self, outfile):
         # do not require quoting, unless explicitly specified, which is necessary for
         # the csc compiler.
         line = line.replace('\\', '/')
+        if mesonlib.is_windows():
+            # Support network paths as backslash, otherwise they are interpreted as
+            # arguments for compile/link commands when using MSVC
+            line = line.replace('//', '\\\\')
```

Comment:
This will replace all double slashes with backslashes. Poking the command line is fragile, so this should instead only replace the first two explicitly. Something like:

```meson
if mesonlib.is_windows() and line.startswith('//'):
    line = '\\\\' + line[2:]
```

Reply:
As explained in #8130, "I would expect multiple slashes to be taken care of upstream by some path normalization".

If you're concerned this is not the case, I can do a replace first of each element in the line variable (as it contains a build statement, so replacing just the start would not work as you suggested).

Updating with this fix shortly.